apple/internal/partials/framework_import.bzl

```diff
@@ -83,12 +92,13 @@ def _framework_import_partial_impl(
             files_to_bundle = [x for x in files_to_bundle if x not in avoid_files]
 
     # Collect the architectures that we are using for the build.
-    build_archs_found = [
-        build_arch
-        for x in targets
-        if AppleFrameworkImportInfo in x
-        for build_arch in x[AppleFrameworkImportInfo].build_archs.to_list()
-    ]
+    build_archs_found_dict = dict()
+    for x in targets:
+        if not AppleFrameworkImportInfo in x:
+            continue
+        for build_arch in x[AppleFrameworkImportInfo].build_archs.to_list():
+            build_archs_found_dict[build_arch] = None
+    build_archs_found = build_archs_found_dict.keys()
```

Comment:
When debugging this PR, I've found that this may contain duplicate values, so I changed the implementation of it. Not sure if I should submit it here.

Reply:
> Not sure if I should submit it here.

Is the change you have here what you aren't sure of? Without the change, what impact does the duplicates have?

Reply:
e.g. `[x86_64, x86_64, x86_64]`

Reply:
The change is kind of unrelated to this PR. I can submit it separately if this PR is going to take some more time. 

Without this, some commands may end up being executed more than once.

Reply:
I would submit it as a followup PR, since it's unrelated to the core of the change.